src/utils.jl

```diff
@@ -69,9 +87,9 @@ for level in [:debug, :info, :warn, :error]
                 # they may expect Logging.shouldlog() getting called, so we use
                 # the global_logger()'s min level which is more likely to be usable.
                 min_level = _invoked_min_enabled_level(global_logger())
-                with_logger(Logging.ConsoleLogger(io, min_level)) do
-                    $(esc(macrocall))
-                end
+                safe_logger = Logging.ConsoleLogger(io, min_level)
+                safe_logstate = Base.CoreLogging.LogState(safe_logger)
+                @safe_with_logstate safe_logstate $(esc(macrocall))
```

Comment:
Why a secondary macro, and not inline the definition here?

If the macro is needed, I think it's also better to do `$GPUCompiler.@safe_with_logstate` to support use of these macros from outside GPUCompiler.jl.

Reply:
No it could totally go inline, it just felt cleaner this way, but I didn't think about the scoping issue when used outside GPUCompiler. Will push an update putting it inline.